include/zephyr/sys/util.h

```diff
@@ -685,6 +685,16 @@ char *utf8_trunc(char *utf8_str);
  */
 char *utf8_lcpy(char *dst, const char *src, size_t n);
 
+/**
+ * @brief Checks if the given string @p str is UTF-8 encoded.
+ *
+ * @param str Target string
+ * @param maxlen The max length of string @p str
+ *
```

Comment:
Maybe there is no need to add a note. Whether it is actually correct depends on the data received, and we have no way to define it as absolutely wrong. This can be solved when the user has the corresponding problem.
(After all, `utf8_trunc` does not give any boundary instructions.)

Reply:
Good point! No validation done [here](https://github.com/zephyrproject-rtos/zephyr/pull/80779/files#diff-ed7182953a91c9e4c191b343f56edd0c475cfcea3ce490399c0f32b8523a0446R36-R43) of the content encoded (i.e. check if appropriate length) done here either!